buildengine/topological.go

```diff
@@ -1,21 +1,22 @@
 package buildengine
 
 import (
+	"fmt"
 	"sort"
 
 	"golang.org/x/exp/maps"
 )
 
-// TopologicalSort returns a sequence of groups of modules in topological order
-// that may be built in parallel.
-func TopologicalSort(graph map[string][]string) [][]string {
+// TopologicalSort attempts to order the modules supplied in the graph based on
+// their topologically sorted order. A cycle in the module dependency graph
+// will cause this sort to be incomplete. The sorted modules are returned as a
+// sequence of `groups` of modules that may be built in parallel. The `unsorted`
+// modules impacted by a dependency cycle are listed in random order.
+func TopologicalSort(graph map[string][]string) (groups [][]string, unsorted []string) {
```

Comment:
what's the reason for returning `unsorted` here instead of the `NewDependencyCycleError` explicitly?

Reply:
currently all the callers next move is to call `NewDependencyCycleError` so returning an error instead of these details might be better - change applied